### PR TITLE
adding vault to brew services

### PR DESCRIPTION
**Pull request by blairham**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/vault.rb [hunk 1]
```diff
@@ -34,6 +34,39 @@ def install
     end
   end
 
+  plist_options :manual => "valt server -dev"
```
Comment (SMillerDev) [inline Formula/vault.rb]: Vault*?

File: Formula/vault.rb [hunk 2]
```diff
@@ -34,6 +34,39 @@ def install
...
+          <string>#{opt_bin}/vault</string>
+          <string>server</string>
+          <string>-dev</string>
+        </array>
+        <key>RunAtLoad</key>
+        <true/>
+        <key>WorkingDirectory</key>
+        <string>#{var}</string>
+        <key>StandardErrorPath</key>
+        <string>#{var}/log/vault.log</string>
```
Comment (SMillerDev) [inline Formula/vault.rb]: This should probably be a separate file.

**Review by SMillerDev — COMMENTED** [review]
.

File: Formula/vault.rb (hunk 1 above)
Comment (blairham) [reply Formula/vault.rb]: good catch, this is the display string and missed the typo

File: Formula/vault.rb (hunk 2 above)
Comment (blairham) [reply Formula/vault.rb]: if you wish, all the others are like this however:

https://github.com/Homebrew/homebrew-core/blob/master/Formula/consul.rb
https://github.com/Homebrew/homebrew-core/blob/master/Formula/nomad.rb

Comment (zbeekman) [reply Formula/vault.rb]: Unless upstream provides a plist file, then this is fine IMO. There are 10s or hundreds of packages, and as far as I know we don't have a policy not to add inline plists or an effort to migrate them some other format.

**Review by zbeekman — APPROVED** [review]
I have confirmed that in-line plists are acceptable provided that upstream does not supply one.

LGTM

**Comment by zbeekman** [comment]
Thanks for your contribution to Homebrew! 🎉 🥇 

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you @blairham!

File: Formula/vault.rb (hunk 2 above)
Comment (blairham) [reply Formula/vault.rb]: excellent I was working on separating this out, but this make it much easier to maintain. Thank you

**Review by zbeekman — COMMENTED** [review]
You may need to add `revision 1` after [line 10](https://github.com/Homebrew/homebrew-core/pull/50584/files#diff-d7846406d4f4249bf301abe5d0185e56R10) to get the plist installed. I'm going to merge this anyway, but we may need a revision bump otherwise the plist/service won't work until the next vault release. I'll report back.

**Comment by zbeekman** [comment]
Just confirmed no revision bump required!

**Pull request closed without merging**